docs/dev/terraform.md

```diff
@@ -1,99 +0,0 @@
-# Embedded Terraform in the Installer
```

Comment:
There are also other docs that have sections mentioning terraform. Do they need to be removed too :thinking:?

- https://github.com/openshift/installer/blob/main/docs/user/metal/install_upi.md#creating-the-cluster
- https://github.com/openshift/installer/blob/main/docs/user/power/install_upi.md#example-power-upi-configuration
- https://github.com/openshift/installer/blob/main/docs/user/vsphere/install_upi.md#example-vsphere-upi-deployment
- https://github.com/openshift/installer/blob/main/docs/dev/azure/azure_client_certs_auth.md#extras

Reply:
Nice, thanks. Let's keep this PR related to build stuff. I will create another story to remove more terraform artifacts (there are still significant amounts).